pkg/dump/ingestor_test.go

```diff
@@ -0,0 +1,662 @@
+					t.Errorf("Dumper.IngestEndpoint() error = %v, wantErr %v", err, tt.wantErr)
+				}
+			}
+		})
+	}
+}
+
+func TestDumpIngestor_IngestRoleBinding(t *testing.T) {
+	t.Parallel()
+	ctx := context.Background()
+
+	// no ingestion
+	noIngest := func(t *testing.T, _ []*rbacv1.RoleBinding) *DumpIngestor {
+		t.Helper()
+		clientset := fake.NewSimpleClientset()
+		dumperIngestor := newFakeDumpIngestor(ctx, t, clientset)
+		m := mockwriter.NewDumperWriter(t)
+		dumperIngestor.writer = m
+
+		return dumperIngestor
+	}
+
+	// ingesting n entries
+	nIngests := func(t *testing.T, roleBindings []*rbacv1.RoleBinding) *DumpIngestor {
+		t.Helper()
+		clientset := fake.NewSimpleClientset()
+		dumperIngestor := newFakeDumpIngestor(ctx, t, clientset)
+		m := mockwriter.NewDumperWriter(t)
+		dumperIngestor.writer = m
+
+		for _, roleDinding := range roleBindings {
+			m.EXPECT().Write(mock.Anything, mock.Anything, ingestRoleBindingPath(roleDinding)).Return(nil).Once()
+		}
+
+		return dumperIngestor
+	}
+
+	type args struct {
+		roleBindings []*rbacv1.RoleBinding
+	}
+	tests := []struct {
+		name           string
+		dumperIngestor *DumpIngestor
+		testfct        func(t *testing.T, roleBindings []*rbacv1.RoleBinding) *DumpIngestor
+		args           args
+		wantErr        bool
+	}{
+		{
+			name:    "no entry",
+			testfct: noIngest,
+			args: args{
+				roleBindings: []*rbacv1.RoleBinding{
+					nil,
+				},
+			},
+			wantErr: true,
+		},
+		{
+			name:    "entries found",
+			testfct: nIngests,
+			args: args{
+				roleBindings: []*rbacv1.RoleBinding{
+					collector.FakeRoleBinding("namespace1", "name1"),
+					collector.FakeRoleBinding("namespace2", "name2"),
+				},
+			},
+			wantErr: false,
+		},
+	}
+	for _, tt := range tests {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+			dumperIngestor := tt.testfct(t, tt.args.roleBindings)
+			for _, roleBinding := range tt.args.roleBindings {
+				if err := dumperIngestor.IngestRoleBinding(ctx, roleBinding); (err != nil) != tt.wantErr {
+					t.Errorf("Dumper.IngestEndpoint() error = %v, wantErr %v", err, tt.wantErr)
+				}
+			}
+		})
+	}
+}
+
+func TestDumpIngestor_IngestRole(t *testing.T) {
+	t.Parallel()
+	ctx := context.Background()
+
+	// no ingestion
+	noIngest := func(t *testing.T, _ []*rbacv1.Role) *DumpIngestor {
+		t.Helper()
+		clientset := fake.NewSimpleClientset()
+		dumperIngestor := newFakeDumpIngestor(ctx, t, clientset)
+		m := mockwriter.NewDumperWriter(t)
+		dumperIngestor.writer = m
+
+		return dumperIngestor
+	}
+
+	// ingesting n entries
+	nIngests := func(t *testing.T, roles []*rbacv1.Role) *DumpIngestor {
+		t.Helper()
+		clientset := fake.NewSimpleClientset()
+		dumperIngestor := newFakeDumpIngestor(ctx, t, clientset)
+		m := mockwriter.NewDumperWriter(t)
+		dumperIngestor.writer = m
+
+		for _, role := range roles {
+			m.EXPECT().Write(mock.Anything, mock.Anything, ingestRolePath(role)).Return(nil).Once()
```

Comment:
(same for all the tests) - this is the thing you should be really testing. I could make this test pass with code that does nothing... You should check that a serialized JSON role is indeed passed to the writer as the core of your test otherwise these tests are not very meaningful